### PR TITLE
Added another test file for Statement test

### PR DESCRIPTION
**Pull request by v-afrafi**
added last test class for statement unit test.

### CODE REVIEW

File: src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java [hunk 1]
```diff
@@ -0,0 +1,2700 @@
...
+public class StatementTest extends AbstractTest {
+    public static final Logger log = Logger.getLogger("StatementTest");
+
+    @Nested
+    public class TCAttentionHandling {
+        private static final int NUM_TABLE_ROWS = 1000;
+        private static final int MIN_TABLE_ROWS = 100;
+        private static final String TEST_STRING = "Hello." + "  This is a test string."
+                + "  It is particularly long so that we will get a multipacket TDS response back from the server." + "  Does it work?"
+                + "  I have no idea." + "  If I did, then I would not be needing this test, now would I?" + "  So how about it then?"
```
Comment (v-ahibr) [inline src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java]: Modify TEST_STRING

File: src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java [hunk 2]
```diff
@@ -0,0 +1,2700 @@
...
+        /**
+         * Test attention sent to interrupt query once it has started execution.
+         *
+         * Expected: Timeout executing request. Verify timeout. Verify subsequent use of connection succeeds.
+         *
+         * Details: The methodology here is similar to our WAITFOR query timeout tests and Defect 58095.
+         */
+        @Test
+        public void testQueryTimeout() throws Exception {
+            // VSTS 433183 - fxUnitStatement: TCAttentionHandling\testQueryTimeout fails on VM
```
Comment (v-ahibr) [inline src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java]: Remove this line

File: src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java [hunk 3]
```diff
@@ -0,0 +1,2700 @@
...
+            // BVTs use a fixed seed (set in build.xml)
```
Comment (v-ahibr) [inline src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java]: this too

File: src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java [hunk 4]
```diff
@@ -0,0 +1,2700 @@
...
+         * Test that cancelling a Statement while consuming a large response ends the response.
+         *
+         * Expected: Response terminates before complete. Verify subsequent use of connection succeeds.
+         *
+         * Details: Test does a large SELECT and expects to get back fewer rows than it asked for after cancelling the ResultSet's associated
+         * Statement.
+         */
+        @Test
+        public void testCancelLongResponse() throws Exception {
+            // assumeTrue("JDBC42".equals(Utils.getConfiguredProperty("JDBC_Version")), "Aborting test case as JDBC version is not compatible. ");
```
Comment (v-ahibr) [inline src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java]: remove un-used code

File: src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java [hunk 5]
```diff
@@ -0,0 +1,2700 @@
...
+                rs = stmt.executeQuery("SELECT * FROM " + tableName);
+
+                // Time how long it takes for execution to be cancelled...
+                long elapsedMillis = -System.currentTimeMillis();
+
+                // Start up a thread to cancel the SELECT after 3 seconds.
+                oneShotCancel = new Thread(new OneShotCancel(stmt, 3));
+                oneShotCancel.start();
+
+                // Start retrieving rows and see how far we get...
```
Comment (v-ahibr) [inline src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java]: change this comment to just "Start retrieving rows"

File: src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java [hunk 6]
```diff
@@ -0,0 +1,2700 @@
...
+            }
+        }
+
+    }
+
+    @Nested
+    public class TCStatementCallable {
+
+        /**
+         * Tests allableStatementMethods on jdbc41
```
Comment (v-ahibr) [inline src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java]: missing "C" in Tests CallableStatementMethods

File: src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java [hunk 7]
```diff
@@ -0,0 +1,2700 @@
...
+            String query = "create procedure " + procName
+                    + " @col1Value varchar(512) OUTPUT, @col2Value varchar(512) OUTPUT AS BEGIN SET @col1Value='hello' SET @col2Value='world' END";
+            stmt.execute(query);
+
+            // Test JDBC 4.1 methods for CallableStatement
+            CallableStatement cstmt = conn.prepareCall("{call " + procName + "(?, ?)}");
+            cstmt.registerOutParameter(1, java.sql.Types.VARCHAR);
+            cstmt.registerOutParameter(2, java.sql.Types.VARCHAR);
+            cstmt.execute();
+            // Temporary solution, verify in the test case. Could not write a wrapper for getObject in fxCallableStatement as fxConnection do not have
```
Comment (v-ahibr) [inline src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java]: remove this comment, should not refer to fx

File: src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java [hunk 8]
```diff
@@ -0,0 +1,2700 @@
...
+            try {
+                stmt.executeUpdate("IF EXISTS (select * from sysobjects where id = object_id(N'" + procName
+                        + "') and OBJECTPROPERTY(id, N'IsProcedure') = 1)" + " DROP PROCEDURE " + procName);
+            }
+            catch (Exception ex) {
+            }
+            ;
+            stmt.executeUpdate("create table " + tableName + " (col1 int, col2 text, col3 int identity(1,1) primary key)");
+            stmt.executeUpdate("Insert into " + tableName + " values(0, 'hello')");
+            stmt.executeUpdate("Insert into " + tableName + " values(0, 'yo')");
```
Comment (v-ahibr) [inline src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java]: avoid strings like yo

File: src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java [hunk 9]
```diff
@@ -0,0 +1,2700 @@
...
+            String query = "create procedure " + procName + " @col1Value int, @col2Value varchar(512) OUTPUT AS BEGIN SELECT * from " + tableName
+                    + " where col1=@col1Value SET @col2Value='wassup' END";
```
Comment (v-ahibr) [inline src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java]: avoid strings like wassup

File: src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java [hunk 10]
```diff
@@ -0,0 +1,2700 @@
...
+         * 
+         * @throws Exception
+         */
+        @Test
+        public void testFailedToResumeTransaction() throws Exception {
+            Class.forName("com.microsoft.sqlserver.jdbc.SQLServerDriver");
+            Connection conn = DriverManager.getConnection(connectionString);
+            Statement stmt = conn.createStatement();
+            try {
+                stmt.executeUpdate("drop table if exists "
```
Comment (v-ahibr) [inline src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java]: avoid using yo

Comment (v-ahibr) [inline src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java]: avoid using wassup

File: src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java [hunk 11]
```diff
@@ -0,0 +1,2700 @@
...
+                        + "') and OBJECTPROPERTY(id, N'IsProcedure') = 1)" + " DROP PROCEDURE " + procName);
+            }
+            catch (Exception ex) {
+            }
+            ;
+            stmt.executeUpdate("create table " + tableName + " (col1 int, col2 text, col3 int identity(1,1) primary key)");
+            stmt.executeUpdate("Insert into " + tableName + " values(0, 'hello')");
+            stmt.executeUpdate("Insert into " + tableName + " values(0, 'yo')");
+            String query = "create procedure " + procName + " @col1Value int, @col2Value varchar(512) OUTPUT AS BEGIN SELECT * from " + tableName
+                    + " where col1=@col1Value SET @col2Value='wassup' END";
```
Comment (v-ahibr) [inline src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java]: do not use strings 'yo' and 'wassap'. may need to retest to make sure nothing broke once strings are modified

File: src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java [hunk 12]
```diff
@@ -0,0 +1,2700 @@
...
+            stmt.execute(query);
+
+            CallableStatement cstmt = conn.prepareCall("{call " + procName + "(?, ?)}");
+            cstmt.setInt(1, 0);
+            cstmt.registerOutParameter(2, java.sql.Types.VARCHAR);
+            ResultSet rs = cstmt.executeQuery();
+            rs.next();
+            assertEquals(rs.getString(2), "hello", "Wrong value");
+            assertEquals(cstmt.getString(2), "wassup", "Wrong value");
```
Comment (v-ahibr) [inline src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java]: avoid using string wassup

File: src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java (hunk 5 above)
Comment (afrafighi) [inline src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java]: done.

**Pull request merged**